velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -166,18 +167,23 @@ class AverageAggregate : public exec::Aggregate {
     if (decodedPartial_.isConstantMapping()) {
       if (!decodedPartial_.isNullAt(0)) {
         auto decodedIndex = decodedPartial_.index(0);
-        auto count = baseCountVector->valueAt(decodedIndex);
-        auto sum = baseSumVector->valueAt(decodedIndex);
-        rows.applyToSelected([&](vector_size_t i) {
-          updateNonNullValue(groups[i], count, sum);
-        });
+        if (!baseSumVector->isNullAt(decodedIndex)) {
+          auto count = baseCountVector->valueAt(decodedIndex);
+          auto sum = baseSumVector->valueAt(decodedIndex);
+          rows.applyToSelected([&](vector_size_t i) {
+            updateNonNullValue(groups[i], count, sum);
+          });
+        }
       }
-    } else if (decodedPartial_.mayHaveNulls()) {
+    } else if (decodedPartial_.mayHaveNulls() || baseSumVector->mayHaveNulls()) {
       rows.applyToSelected([&](vector_size_t i) {
         if (decodedPartial_.isNullAt(i)) {
           return;
         }
         auto decodedIndex = decodedPartial_.index(i);
+        if (baseSumVector->isNullAt(decodedIndex)) {
+          return;
+        }
```

Comment:
If this fix is reasonable, we can also add it to `addSingleGroupIntermediateResults`.